presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockBuilder.java

```diff
@@ -295,6 +295,37 @@ public BlockBuilder appendSingleStructure(Block block)
         return this;
     }
 
+    @Override
+    public BlockBuilder appendStructure(Block block, int position)
+    {
+        if (!(block instanceof AbstractMapBlock)) {
+            throw new IllegalArgumentException();
+        }
+
+        beginBlockEntry();
```

Comment:
Don't use beginBlockEntry. The protocol of beginBlockEntry is that you operate on the returned entry BlockBuilder once you call it until you close the entry. Operating directly on the original BlockBuilder after calling `beginBlockEntry` is a bad idea.